pkgs/development/libraries/numcpp/default.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  boost181,
+  python3,
+  gtest,
+  eigen,
+  opencv,
+}:
+let
+  boost = boost181.override { enableStatic = true; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "numcpp";
+  version = "2.11.0";
+
+  src = fetchFromGitHub {
+    owner = "dpilger26";
+    repo = "NumCpp";
+    rev = "Version_${finalAttrs.version}";
+    hash = "sha256-IAku1bcaMkawZxpQbvxcS6VX07ogw4UGo1DX2Wa8xwU=";
+  };
+
+  nativeBuildInputs = [cmake gtest python3 eigen opencv];
+
+  cmakeFlags = ["-DBUILD_ALL=ON"];
+
+  buildInputs = [boost];
+
+  postInstall = ''
+    substituteInPlace $out/share/NumCpp/cmake/NumCppConfig.cmake \
+      --replace "\''${PACKAGE_PREFIX_DIR}/" ""
+  '';
+
+  NIX_CFLAGS_COMPILE="-Wno-error";
+
```

```suggestion
  doCheck = true;
```

The check phase needs to be enabled for the tests to actually be run after they're built.

---

It might be necessary to set disable for this if `stdenv.isDarwin` - I can't get the tests to build on macos 10.15:

```
/tmp/nix-build-numcpp-2.11.0.drv-0/source/test/pytest/src/Functions.cpp:2678:16: error: call to 'signbit' is ambiguous
        return signbit(inValue);
               ^~~~~~~
/tmp/nix-build-numcpp-2.11.0.drv-0/source/test/pytest/src/Functions.cpp:3733:49: note: in instantiation of function template specialization 'FunctionsInterface::signbitScalar<double>' requested here
    m.def("signbitScalar", &FunctionsInterface::signbitScalar<double>);
                                                ^
/nix/store/nq85mb7p7fwcm1cad3af02i3kdmsybii-libcxx-11.1.0-dev/include/c++/v1/math.h:329:1: note: candidate function [with _A1 = double]
signbit(_A1 __lcpp_x) _NOEXCEPT
^
/tmp/nix-build-numcpp-2.11.0.drv-0/source/include/NumCpp/Functions/signbit.hpp:46:10: note: candidate function [with dtype = double]
    bool signbit(dtype inValue) noexcept
```

:shrug:

---

i made it so it doesnt build if you are using darwin

---

I think it also needs to disable tests if `stdenv.hostPlatform.isStatic` - the tests don't appear to be designed to build on a static toolchain.

---

added that too